visualization/view_scene.py

```diff
@@ -0,0 +1,111 @@
+"""
+Script to render a GTSFM scene using either Open3d or Mayavi mlab.
+
+Author: John Lambert
+"""
+
+import argparse
+import os
+from pathlib import Path
+
+import numpy as np
+from gtsam import Rot3, Pose3
+
+import gtsfm.utils.io as io_utils
+
+from visualization.open3d_vis_utils import draw_scene_open3d
+from visualization.mayavi_vis_utils import draw_scene_mayavi
+
+REPO_ROOT = Path(__file__).parent.parent.resolve()
+
+
+def compute_point_cloud_center_robust(point_cloud: np.ndarray) -> np.ndarray:
+    """Robustly estimate the point cloud center.
+
+    Args:
+        point_cloud: array of shape (N,3) representing 3d points.
+
+    Returns:
+        mean_pt: coordinates of central point, ignoring outliers.
+    """
+    ranges = np.linalg.norm(point_cloud, axis=1)
```

Comment:
is there some reference for this?

Reply:
so the main idea was to just that the worst outliers were far-away points. So i restrict the range when estimating the center of the point cloud.